### DISCUSSION TITLE
Managing Multiple Auths

### DISCUSSION THREAD

**Discussion by matthewfkoch** [prometheus/snmp_exporter#1178] [Q&A]
I currently have two pools of devices being managed which the main difference being the Auth they are programmed to use. The challenge is as devices are updated they changed from using Auth A to Auth B and they will fail until I update the configuration in Prometheus. I'm wondering if there is a way to specify more than one Auth in Prometheus so as devices are upgraded, we don't lose data.

**Comment by SuperQ** [comment]
You could always add all devices with both auths as two different jobs. You would have ignore scrape failures for one of two auths tho.

Say you have `job_name: snmp_auth_a` and `job_name: snmp_auth_b`.

You could make sure that at least one scrape is succeeding with:
```
sum without (job) (up{job=~"snmp_auth.*"}) > bool 0
```

**Reply by matthewfkoch** [reply]
The challenge with this method would be that the job with the incorrect auth but a valid device will continue to scrape the device with the bad credentials which will lead to unnecessary polling.

**Reply by SuperQ** [reply]
Yea, but the polling will get blocked/ignored by the device, which should be zero load.

On the Prometheus / exporter side, the extra polling is very low load.

The only good way to solve this is to use a dynamic discovery database. For example, if you had the configuration synced with Ansible, or had a tool that would update the auth information in Netdata.

Prometheus can then update within seconds of the discovery data changing.
